common/main/java/com/couchbase/lite/MutableArray.java

```diff
@@ -38,41 +38,45 @@
     /**
      * Constructs a new empty Array object.
      */
-    public MutableArray() {
+    public MutableArray() { super(); }
+
+    /**
+     * Creates a new MutableArray with content from the passed List.
+     * Allowed value types are List, Date, Map, Number, null, String, Array, Blob, and Dictionary.
+     * The List and Map must contain only the above types.
+     *
+     * @param data the document content list
+     */
+    public MutableArray(@NonNull List<Object> data) {
         super();
+        setData(data);
     }
 
     /**
-     * Constructs a new Array object with an array content. Allowed value types are List, Date,
-     * Map, Number, null, String, Array, Blob, and Dictionary. The List and Map must contain
-     * only the above types.
+     * Creates a new MutableArray with content from the passed JSON string.
      *
-     * @param data the array object.
+     * @param json the array content as a JSON string.
      */
-    public MutableArray(List<Object> data) {
+    public MutableArray(@NonNull String json) {
         super();
-        setData(data);
+        setJSON(json);
     }
 
     // to create copy
-    MutableArray(MArray mArray, boolean isMutable) {
-        super(mArray, isMutable);
-    }
+    MutableArray(MArray mArray, boolean isMutable) { super(mArray, isMutable); }
 
     // Call from native method
-    MutableArray(MValue mv, MCollection parent) {
-        super(mv, parent);
-    }
+    MutableArray(MValue mv, MCollection parent) { super(mv, parent); }
 
     //---------------------------------------------
     // API - public methods
     //---------------------------------------------
 
     /**
-     * Set an array as a content. Allowed value types are List, Date,
-     * Map, Number, null, String, Array, Blob, and Dictionary. The List and Map must contain
-     * only the above types. Setting the new array content will replace the current data
-     * including the existing Array and Dictionary objects.
+     * Populate an array with content from a Map.
```

Comment:
confused! you mean Map or List? Argument is `List<Object>`?

Reply:
Yeah... it is not totally clear.  The idea is that the List can contains Maps or Lists.  If it does, those Maps/Lists must contain only the types mentioned above.  I've revisited the comments to try to make them clearer.